### PR TITLE
Vue3 Documentation

### LINKED ISSUE #2878: [Feedback]Tutorial doesn't seems to work with Vue3
**Page**: [`/start/getting-started/setup/q/integration/vue`](https://docs.amplify.aws/start/getting-started/setup/q/integration/vue)

**Feedback**:

Once I get to the step of [setting up frontend here](https://docs.amplify.aws/start/getting-started/setup/q/integration/vue#set-up-frontend), I get this error when running the server: 
```
Uncaught TypeError: Cannot read property 'config' of undefined
```

The command fails at this line: `Vue.config.ignoredElements = [/amplify-\w*/];` 
Although the syntax in the tutorial seems to be Vue3 syntax.

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:* Fixes #2878

*Description of changes:* This updates the getting started guide and ui component docs to reflect Vue 3 release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/start/getting-started/fragments/vue/setup.md [hunk 1]
```diff
@@ -240,10 +240,22 @@ Open __src/main.js__ and add the following code below the last import:
...
+applyPolyfills().then(() => {
+  defineCustomElements(window);
+});
 ```
 
 Now Amplify has been successfully configured. As you add or remove categories and make updates to your backend configuration using the Amplify CLI, the configuration in __aws-exports.js__ will update automatically.
+
+<amplify-callout>
+
+Console warnings: if you see "failed to resolve component" warnings, you can create `vue.config.js` from the app directory and use this <amplify-external-link href="https://gist.github.com/wlee221/3d47f9598d5ad85bfa7a138bad112c3c">gist</amplify-external-link> to remove them.
```
Comment (sammartinez) [inline docs/start/getting-started/fragments/vue/setup.md:259]:
```suggestion
Console warnings: If you see "failed to resolve component" warnings, you can create `vue.config.js` from the app directory and use this <amplify-external-link href="https://gist.github.com/wlee221/3d47f9598d5ad85bfa7a138bad112c3c">gist</amplify-external-link> to remove them.
```

Comment (sammartinez) [inline docs/start/getting-started/fragments/vue/setup.md:259]:
```suggestion
Console warnings: If you see "failed to resolve component" warnings, you can create `vue.config.js` from the app directory and use this <amplify-external-link href="https://gist.github.com/wlee221/3d47f9598d5ad85bfa7a138bad112c3c">gist</amplify-external-link> to remove them.
```

File: docs/ui/auth/fragments/web/authenticator.md [hunk 2]
```diff
@@ -860,6 +860,35 @@ _App.vue_
...
+<amplify-block-switcher>
+<amplify-block name="Vue 3">
+
+```js
+import {onAuthUIStateChange} from "@aws-amplify/ui-components";
+
+export default {
+  name: "AuthStateApp",
+  created() {
+    onAuthUIStateChange((authState, authData) => {
```
Comment (ericclemmons) [inline docs/ui/auth/fragments/web/authenticator.md:872]: Should this also have the `Auth.currentAuthenticatedUser().then((user) => this.user = user).catch(console.warn)` block?

**Review by ericclemmons — COMMENTED** [review]
Looking good so far!

I added some comments based on our bug bash discussions yesterday around `onAuthUIStateChange`.

File: docs/ui/auth/fragments/web/authenticator.md (hunk 2 above)
Comment (wlee221) [reply docs/ui/auth/fragments/web/authenticator.md:872]: Based on our latest discussion, I'll work on resolving this from the library side.

File: docs/ui/auth/fragments/web/authenticator.md [hunk 3]
```diff
@@ -860,6 +860,35 @@ _App.vue_
...
+    });
+  },
+  data() {
+    return {
+      user: undefined,
+      authState: undefined,
+    };
+  },
+  beforeUnmount() {
+    return onAuthUIStateChange;
```
[the remark this answers is not in the document]
Comment (wlee221) [reply docs/ui/auth/fragments/web/authenticator.md:884]: You're right, will do! Angular has the same problem but I'll deal with Angular on another PR.

[the remark this answers is not in the document]
Comment (wlee221) [reply docs/ui/auth/fragments/web/authenticator.md]: Updated 👍

**Review by sammartinez — APPROVED** [review]
LGTM 🌮 . Let's get the changes into release prior to us going live with these docs

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**